agent/api/task/task.go

```diff
@@ -846,6 +876,81 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 	return hostConfig, nil
 }
 
+func (task *Task) awslogsEndpointOverride(hostConfig *docker.HostConfig, dockerVersion string) bool {
+	if hostConfig.LogConfig.Type != awslogsType {
+		return false
+	}
+
+	isVersionMatch := false
+	if task.isDockerCEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerCEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else if task.isDockerEEVersion(dockerVersion) {
+		match, err := utils.Version(dockerVersion).Matches(">=" + awslogsEndpointMinimumDockerEEVersion)
+		if err != nil {
+			seelog.Warnf("Unable to compare docker version: %v", err)
+		} else if match {
+			isVersionMatch = true
+		}
+	} else {
+		seelog.Warnf("Docker version %s is neither CE version or EE version", dockerVersion)
+	}
+
+	if isVersionMatch {
+		return hostConfig.LogConfig.Config[awslogsEndpointOpt] == ""
+	} else {
+		// If the Docker version is not matched, make the awslogs endpoint option empty
+		// as it's not supported
+		if hostConfig.LogConfig.Config[awslogsEndpointOpt] != "" {
+			hostConfig.LogConfig.Config[awslogsEndpointOpt] = ""
+		}
+	}
+
+	return false
+}
+
+func (task *Task) isDockerCEVersion(dockerVersion string) bool {
```

Comment:
I'd like you to remove these functions anyway (see above for an alternate suggestion), but for future reference: the Docker version is not a property of the task, so it shouldn't have a `(task *Task)` receiver.